### PR TITLE
improve tests mmm utils

### PR DESCRIPTION
**Pull request by juanitorduz**
Remove unused code and add more tests

----
📚 Documentation preview 📚: https://pymc-marketing--738.org.readthedocs.build/en/738/

### CODE REVIEW

File: pymc_marketing/mmm/utils.py [hunk 1]
```diff
@@ -227,32 +226,6 @@ def find_sigmoid_inflection_point(
     return x_inflection, y_inflection
 
 
-def standardize_scenarios_dict_keys(d: dict, keywords: list[str]):
```
Comment (juanitorduz) [inline pymc_marketing/mmm/utils.py:230]: This function is never used and did not have a test

Comment (cetagostini) [reply pymc_marketing/mmm/utils.py:230]: Indeed legacy from the previous optimizer. Good catch 🙏🏻

File: tests/mmm/test_utils.py [hunk 2]
```diff
@@ -344,3 +346,33 @@ def test_create_new_spend_data(
...
+    ],
+    ids=["Normal", "Gamma", "StudentT"],
+)
+def test_get_distribution_from_dict(distribution_dict, expected):
+    assert _get_distribution_from_dict(distribution_dict) == expected
+
+
+def test_get_distribution_from_dict_value_error():
+    with pytest.raises(ValueError):
+        _get_distribution_from_dict({"dist": "InvalidDistribution"})
```
Comment (wd60622) [inline tests/mmm/test_utils.py:378]: There are some tests in tests/mmm/test_delayed_saturation_mmm.py for this 
Let's delete those and go with theses

Comment (juanitorduz) [reply tests/mmm/test_utils.py:378]: looking into it, it seems we have duplicated code anyway https://github.com/pymc-labs/pymc-marketing/pull/738/commits/e6067e04a6a68052a96c28cc6cabbc3de8fc088d

**Review by wd60622 — APPROVED** [review]
LGTM!

**Pull request merged**